napari/_vispy/_tests/test_vispy_labels_layer.py

```diff
@@ -0,0 +1,135 @@
+            labels = zarr.open(
+                fn, shape=(20, 20), dtype=np.uint32, chunks=(10, 10)
+            )
+        if array_type == 'tensorstore':
+            ts = pytest.importorskip('tensorstore')
+            spec = {
+                'driver': 'zarr',
+                'kvstore': {'driver': 'file', 'path': fn},
+                'path': '',
+                'metadata': {
+                    'dtype': labels.dtype.str,
+                    'order': labels.order,
+                    'shape': labels.shape,
+                },
+            }
+            labels = ts.open(spec, create=False, open=True).result()
+
+        layer = viewer.add_labels(labels)
+        QCoreApplication.instance().processEvents()
+        layer.paint((10, 10), 1, refresh=True)
+        visual = viewer.window._qt_viewer.layer_to_visual[layer]
+        assert np.any(visual.node._data)
+
+
+@skip_local_popups
+@pytest.mark.parametrize('array_type', ['numpy', 'zarr', 'tensorstore'])
+def test_labels_fill_slice(make_napari_viewer, array_type):
+    """Check that painting labels paints only on current slice.
+
+    This should work regardless of array type. See:
+    https://github.com/napari/napari/issues/6079
+    """
+    viewer = make_napari_viewer(show=True)
+    with tempfile.TemporaryDirectory(suffix='.zarr') as fn:
+        if array_type == 'numpy':
+            labels = np.zeros((3, 20, 20), dtype=np.uint32)
+        elif array_type in {'zarr', 'tensorstore'}:
+            labels = zarr.open(
+                fn, shape=(3, 20, 20), dtype=np.uint32, chunks=(1, 10, 10)
+            )
+        if array_type == 'tensorstore':
+            ts = pytest.importorskip('tensorstore')
+            spec = {
+                'driver': 'zarr',
+                'kvstore': {'driver': 'file', 'path': fn},
+                'path': '',
+                'metadata': {
+                    'dtype': labels.dtype.str,
+                    'order': labels.order,
+                    'shape': labels.shape,
+                },
+            }
+            labels = ts.open(spec, create=False, open=True).result()
+
+        labels[0, :, :] = 1
+        labels[1, 10, 10] = 1
+        labels[2, :, :] = 1
+        layer = viewer.add_labels(labels)
+        layer.n_edit_dimensions = 3
+        QCoreApplication.instance().processEvents()
+        layer.fill((1, 10, 10), 13, refresh=True)
+        visual = viewer.window._qt_viewer.layer_to_visual[layer]
+        assert np.sum(visual.node._data) == 13
+
+
+@skip_local_popups
+@pytest.mark.parametrize('array_type', ['numpy', 'zarr', 'tensorstore'])
+def test_labels_painting_with_mouse(
+    MouseEvent, make_napari_viewer, array_type
+):
+    """Check that painting labels paints on the canvas when using mouse.
+
+    This should work regardless of array type. See:
+    https://github.com/napari/napari/issues/6079
+    """
+    viewer = make_napari_viewer(show=True)
+    with tempfile.TemporaryDirectory(suffix='.zarr') as fn:
```

Comment:
@Czaki I've always been a bit wary of writing my own fixtures because I see them as too magical. (It's impossible as a Python newcomer for me to tell where `tmp_path` comes from, for example, though I am happy to use it — does make the test better.) So I'll make a normal function here. 🙏

Reply:
it is core pytest feature https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
And I like it as it make code more readable for me.

Reply:
It is extremely non-local behaviour. Looking at this file, there's literally no way for me to know (without running the test) whether it should be tmp_path or temp_path. And in fact maybe temp_path is a different fixture provided by a plugin with subtly different behaviour... So now a simple typo can result in some hard-to-trace error in tests. If one had to import fixtures explicitly (for example, they could be default values for the parameters), this problem would be avoided.

Reply:
but this will allow to rename etc. Now I know that `tmp_path` is `tmp_path` and I do not nned to think why `tmp_path` in one test behaves differently the same name in another (with small exceptions). 

But yes, it is excellent if you are familiar with it, but could be strange when you do not use it too often.